deploy/sre-prometheus/100-managed-velero-operator.PrometheusRule.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: sre-managed-velero-operator-alerts
+    role: alert-rules
+  name: sre-managed-velero-operator-alerts
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-managed-velero-operator-alerts
+    rules:
+    - alert: VeleroHourlyObjectBackupMissed
+      expr: |
+        time() - clamp_min(velero_backup_last_successful_timestamp{namespace="openshift-velero",schedule="hourly-object-backup"},
```

Comment:
so if I understand this correctly:
current time - (the lower of the last successful backup, OR when the deployment was created): in a working cluster, this should be somewhere between 0 and 3600 (one hour)

so we take that, and measure if it's greater than

3600 + the average backup duration for the last 5 hours

This is really cool. Two thoughts.. 
Should we add a buffer? Say like a static extra 5 mins.. or some added percentage above the average.. just in case it's slightly longer than the average?
Would it be good to also clamp that average to a max, so if it can never go over a certain amount (say a static 30 minutes on each of these jobs)?

Reply:
You got it right except the average backup time itself acts as the buffer, because `velero_backup_last_successful_timestamp` marks when the last backup _finished_, not started.  So this actually gives it roughly the span of _two_ backup durations to finish: the hourly backup should be _finishing_ about 3600 seconds after the last one finished (since the durations are pretty consistent), plus the average time tacked on as a buffer.

The average calculation might be overkill though if, like you suggest, a constant buffer would suffice.  Or we could tighten the buffer and make it a fraction of the average, say like `3600 + (AVERAGE * 0.25)`.  Also good idea on clamping the average.

Sounds like it's just a matter of tuning now.

Reply:
Ohhh okay. That makes sense.

My gut says it probably doesn't have to be that complex. Right now the average time for a base cluster is about 60 seconds.. if we give a static buffer of 5 or 10 minutes from the last complete time, that might be enough. That way, we don't have to clamp it, it's just a static number + the last completion time.